src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -108,7 +109,16 @@ public void run(ActionContext context) throws Exception {
     LOG.debug("The BigQuery SQL is {}", config.getSql());
 
     // Wait for the query to complete
-    queryJob.waitFor();
+    try {
+      queryJob.waitFor();
+    } catch (BigQueryException e) {
+      // Minor bug with BigQuery java client causing it to prematurely throw an exception
+      // Pipeline should always fail if an exception is encountered, but BigQueryErrors take priority
+      // See CDAP-17061
+      if (queryJob == null || queryJob.getStatus() == null || queryJob.getStatus().getError() == null) {
+        throw new RuntimeException(e);
```

Comment:
BigQueryException is already a RuntimeException so it can be thrown as-is instead of wrapping it in a RuntimeException. In most cases, it's better to propagate runtime exceptions without wrapping them

Reply:
Fixed